src/world_new/Robot.cpp

```diff
@@ -88,6 +88,14 @@ bool Robot::isWorkingBallSensor() const noexcept { return workingBallSensor; }
 
 void Robot::setWorkingBallSensor(bool _workingBallSensor) noexcept { Robot::workingBallSensor = _workingBallSensor; }
 
+bool Robot::hasBallBallSensor() const noexcept { return hasBall; }
```

Comment:
The naming is a bit weird, maybe something like this:
```suggestion
bool Robot::BallSensorSeesBall() const noexcept { return hasBall; }
```

Reply:
Also the implementation does not make sense to me. What is hasBall? There is a function with that name as well in RobotView, that does something different.

Reply:
Done